### PR TITLE
Clarify add_done_callback works on both success and failure

### PR DESCRIPTION
**Pull request by mehrdada**

### CODE REVIEW

File: src/python/grpcio/grpc/__init__.py [hunk 1]
```diff
@@ -170,7 +170,8 @@ def traceback(self, timeout=None):
 
     @abc.abstractmethod
     def add_done_callback(self, fn):
-        """Adds a function to be called at completion of the computation.
+        """Adds a function to be called at completion of the computation,
+        whether or not the computation is terminated sucessfully.
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/grpc/__init__.py]: ["A doc string should be organized as a summary line (one physical line) terminated by a period, question mark, or exclamation point, followed by a blank line, followed by the rest of the doc string starting at the same cursor position as the first quote of the first line."](https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments)

This is good content to add; it just doesn't fit into the one-line summary. Put it in one of the arbitrary-length paragraphs that follows the one-line summary.

**Comment by mehrdada** [comment]
I am not confident about the test coverage at all--However that is orthogonal to clarification of the spec, which in my understanding is the way we intend this to work. Are you suggesting that it is not?

File: src/python/grpcio/grpc/__init__.py (hunk 1 above)
Comment (mehrdada) [reply src/python/grpcio/grpc/__init__.py]: Thanks. Done

**Comment by mehrdada** [comment]
Cool, thanks for the link. I am glad that is the intended behavior, because otherwise it would be a pretty weird future implementation. Considering the documentation of `concurrent.futures.Future` explicitly spells out the behavior too, and that non-zero number of people have been confused by this, I am inclined to say it is worthy of addition.

**Pull request merged**